localMatchApp/sources/homewindow.cpp

```diff
@@ -48,3 +64,65 @@ HomeWindow::HomeWindow ( QWidget *i_parent ) : QDialog ( i_parent ), ui ( new Ui
 HomeWindow::~HomeWindow ( ) {
   delete ui;
 }
+
+void HomeWindow::onButtonPreviousClicked()
+{
+    index--;
+    Functions f;
+    Events event = f.SelectEvent(index);
+    eventStatus = QString::number(index+OPERATIONONE)+"/"+QString::number(event.getTotalEvents())+" of events";
+    ui->labelEventStatus->setText(eventStatus);
+    if(event.getTotalEvents() == NOEVENTS){
+        ui->labelNameEvent->setText("No Data");
+    }else if (event.getTotalEvents() == ONEEVENT) {
+        ui->ButtonPrevious->setDisabled(true);
+        ui->ButtonNext->setDisabled(true);
+    }else if(event.getTotalEvents() > ONEEVENT){
+        if(index==FIRSTEVENT){
+            ui->ButtonPrevious->setDisabled(true);
+        }else{
+            ui->ButtonPrevious->setDisabled(false);
+        }
+        if(index==(event.getTotalEvents()-OPERATIONONE)){
+            ui->ButtonNext->setDisabled(true);
+        }else{
+            ui->ButtonNext->setDisabled(false);
+        }
+    }
+    ui->labelNameEvent->setText (event.getNameEvent());
+    ui->labelDescription->setText (event.getDescriptionEvent());
+    ui->labelEventInitialDate->setText (event.getInitialDateEvent());
+    ui->labelEventFinalDate->setText (event.getFinalDateEvent());
+    ui->pictureEvent->setPixmap (event.getPathImageEvent());
+}
+
+void HomeWindow::onButtonNextClicked()
```

Comment:
Refactor this two functions, the code is repeated and too extensive, difficult to read